torch/csrc/api/src/nn/modules/linear.cpp

```diff
@@ -79,6 +80,56 @@ Tensor FlattenImpl::forward(const Tensor& input) {
 
 // ============================================================================
 
+UnflattenImpl::UnflattenImpl(const UnflattenOptions& options_) : options(options_) {
+  reset();
+}
+
+void UnflattenImpl::reset() {
+  named = c10::get_if<std::string>(&options.dim()) != nullptr;
```

Comment:
In this case, we don't even need a boolean `named` right? 
We can use this check directly in  forward and pretty_print.